test/boost/schema_change_test.cc

```diff
@@ -637,7 +637,13 @@ future<> test_schema_digest_does_not_change_with_disabled_features(sstring data_
         }
 
         auto expect_digest = [&] (schema_features sf, utils::UUID expected) {
-            auto actual = calculate_schema_digest(service::get_storage_proxy(), sf).get0();
+            // Changes in the system_distributed keyspace was a common source of regenerating this test case.
+            // It was not the original idea of this test to regenerate after each change in the distributed
+            // system keyspace, since these changes are propagated naturally on their own anyway - the schema
+            // with highest timestamp will win and be sent to other nodes.
+            // Thus, system_distributed.* tables are officially not taken into account,
+            // which makes it less likely that this test case would need to be needlessly regenerated.
+            auto actual = calculate_schema_digest(service::get_storage_proxy(), sf, &is_internal_keyspace).get0();
```

Comment:
Maybe a more positive way is to specify the keyspace for our test tables (`tests`). This way we don't depend on the muddy definition of is_internal_keyspace.

On the other hand, a positive way means that if we misspell the keyspace name, the test does nothing.

Reply:
(A positive way could even be exposed: calculate a digest for this keyspace with a REST API or cql function; this way we could run the tests manually for a selected schema:

```sql
cql> SELECT keyspace_schema_digest('my_keyspace')
blah-blah-blah
```

So you can pre-test interactively before going to schema_change_test to formalize the test.

Reply:
On the other hand, if somebody adds an internal keyspace and forgets to mark it as internal, this test will incidentally catch that and report a failed schema. So perhaps it still makes sense to digest all non-internal tables here.